### PR TITLE
Add 3 functions for getting the display mode

### PR DESCRIPTION
**Pull request by bastienleonard**
* `get_desktop_display_mode()`
* `get_current_display_mode()`
* `Window.display_index`

(Maybe the first two functions should go in a new `video` file? SDL's wiki lists them under the Video category.)

### CODE REVIEW

**Comment by bastienleonard** [comment]
Reworked the commit and MR to include Window.display_index, which is necessary to use the other two functions.

File: src/sdl.cr [hunk 1]
```diff
@@ -66,4 +66,16 @@ module SDL
       LibSDL.quit
     end
   end
+
+  def self.get_desktop_display_mode(display_index : Int32) : LibSDL::DisplayMode
+    ret = LibSDL.get_desktop_display_mode(0, out mode)
```
Comment (ysbaddaden) [inline src/sdl.cr:71]:
```suggestion
    ret = LibSDL.get_desktop_display_mode(display_index, out mode)
```

File: src/sdl.cr [hunk 2]
```diff
@@ -66,4 +66,16 @@ module SDL
...
+    raise Error.new("SDL_GetDesktopDisplayMode") unless ret == 0
+    mode
+  end
+
+  def self.get_current_display_mode(display_index : Int32) : LibSDL::DisplayMode
+    ret = LibSDL.get_current_display_mode(0, out mode)
```
Comment (ysbaddaden) [inline src/sdl.cr:77]:
```suggestion
    ret = LibSDL.get_current_display_mode(display_index, out mode)
```

**Review by ysbaddaden — CHANGES_REQUESTED** [review]
What about dropping the `get_` prefixes, so that we'd have `SDL.desktop_display_mode` and `SDL.current_display_mode`?

I'm wondering whether there could be a single function and a DisplayMode enum (Desktop, Current), so we'd call `SDL.display_mode(:current, 0)` and `SDL.display_mode(:desktop, 0)` for example, but that's probably overkill.

**Comment by bastienleonard** [comment]
I made the changes that you requested. I was hoping that GitHub would update the diffs, but that doesn't to be the case.

>What about dropping the get_ prefixes, so that we'd have SDL.desktop_display_mode and SDL.current_display_mode?

Personally I generally prefer that functions don't have names that would be typical of attributes. But it depends on the convention you're using for the project, so it's your call.

>I'm wondering whether there could be a single function and a DisplayMode enum (Desktop, Current), so we'd call SDL.display_mode(:current, 0) and SDL.display_mode(:desktop, 0) for example, but that's probably overkill.

Personally I find that SDL2 is well designed overall for a C library and I prefer when bindings just adapt to the target language's idions. For example, in Rust I prefer to use the sys library that is just an unsafe wrapper around the C API, rather the full-fledged API which changes everything for no apparent reason.

**Comment by ysbaddaden** [comment]
I'm following Ruby/Crystal practice of avoiding the get/set/is/has prefixes when possible:

- `get_name` -> `name`
- `set_name` -> `name=`
- `has_name` -> `name?`

For example `window.cr` is full of these.

**Comment by bastienleonard** [comment]
If the goal is to make the method look like an attribute, then that seems in conflict with the fact that it takes a parameter (the display index).

I have added a commit with an alternative API. Typical usage would become `SDL.desktop_display_modes[window.display_index]`. One downside is that it makes the API slightly magical, which will confuse users if the abstraction leaks (e.g. in case of SDL error, users will have to realize it's not an actual array).

**Comment by bastienleonard** [comment]
Sounds good, I removed the proposal commit and added a new one where I started working on the new API.

**Comment by bastienleonard** [comment]
Can you give me feedback on the last commit?

**Review by ysbaddaden — APPROVED** [review]

**Comment by ysbaddaden** [comment]
Thanks!

**Pull request merged**